components/store/CartUpdateDialog.js

```diff
@@ -0,0 +1,224 @@
+
+    return (
+      <Dialog
+        className="cart-update-dialog"
+        controls={updateComplete ? undefined : this.controls}
+        title="Add to Cart"
+        onClose={onClose}>
+        <div className="center-content">
+          {do {
+            if (!updateComplete) {
+              <div className="product-card cart-view">
+                {images && images[0] && (
+                  <img src={images[0]} alt="product preview" />
+                )}
+                <span className="title">
+                  {name}
+                </span>
+                <p>{caption}</p>
+                <div className="controls">
+                  {(prodAttr && prodAttr[0]) && (
+                    <div className="labeled-input">
+                      <label htmlFor="skuSelect">
+                        {prodAttr[0].replace(/^\w/, char => char.toUpperCase())}:
+                      </label>
+                      <select
+                        id="skuSelect"
+                        name="activeSKU"
+                        value={this.state.activeSKU}
+                        onChange={this._handleSKUChange}>
+                        {Object.values(skus).map(sku => {
+                          const inStock = isInStock(sku.inventory)
+                          return <option value={sku.id} key={sku.id} disabled={!inStock}>{sku.attributes[prodAttr[0]]} {!inStock && '(Sold Out!)'}</option>
+                        })}
+                      </select>
+                    </div>
+                  )}
+
+                  <div className="labeled-input">
+                    <label htmlFor="quantitySelect">Quantity:</label>
+                    <input
+                      disabled={activeSKU && !isInStock(activeSKU.inventory)}
+                      id="quantitySelect"
+                      name="quantity"
+                      onChange={this._handleQuantityChange}
+                      type="number"
+                      min={0}
+                      max={activeSKU.inventory.quantity || undefined}
+                      value={this.state.quantity} />
+                  </div>
+                </div>
+              </div>
+            } else {
+              <div>
+                <h1>{product.attributes.name} Updated!</h1>
+              </div>
+            }
+          }}
+        </div>
+      </Dialog>
+    )
+  }
+
+  get controls () {
+    const {
+      cart,
+      product,
+    } = this.props
+    const {
+      activeSKU: aSKU,
+      quantity,
+    } = this.state
+
+    const activeSKU = aSKU !== '' ? product.attributes.skus[aSKU] : null
+
+    return {
+      primary: [
+        (
+          <span>
+            {
+              activeSKU && quantity > 0
+                ? `${quantity} @ ${(activeSKU.price / 100).toLocaleString(...currencyStringOptions)}${quantity > 1 ? ` = ${((activeSKU.price * quantity) / 100).toLocaleString(...currencyStringOptions)}` : ''}`
```

Comment:
This is horribly unreadable, we can let it pass for now but there should be a note to refactor this